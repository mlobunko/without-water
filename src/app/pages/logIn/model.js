export const init = objBody => ({
  method: 'POST',
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json'
  },
  body: JSON.stringify(objBody)
});

export const validator = (email, password) => {
  const emailExp = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  const emailValidate = emailExp.test(email);

  const passwordExp = /^\w{5,16}$/;
  const passwordValidate = passwordExp.test(password);

  return {
    email: emailValidate,
    password: passwordValidate
  };
};
