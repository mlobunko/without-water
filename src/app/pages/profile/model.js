import * as icons from './icons';

export const getProfile = id =>
  `https://mysterious-reef-29460.herokuapp.com/api/v1/user-info/${id}`;

export const placeWebIconFirst = arr => {
  return arr.sort(el => {
    return el.label === 'web' ? -1 : 1;
  });
};

export const getIconByName = iconName => {
  switch (iconName) {
    case 'telegram':
      return icons.IconTelegram;
    case 'vk':
      return icons.IconVk;
    case 'twitch':
      return icons.IconTwitch;
    case 'twitter':
      return icons.IconTwitter;
    case 'web':
      return icons.IconGlobe;
    case 'youtube':
      return icons.IconYoutube;
    default:
      return undefined;
  }
};
