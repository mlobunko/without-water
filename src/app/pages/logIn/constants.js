export const NAME = 'logIn';

export const POST_REQUEST =
  'https://mysterious-reef-29460.herokuapp.com/api/v1/validate';

export const ERROR_TRANSLATE = {
  wrong_email_or_password: 'Wrong email or password'
};
