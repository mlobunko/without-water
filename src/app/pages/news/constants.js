export const NAME = 'news';

export const REQUEST_NEWS =
  'https://mysterious-reef-29460.herokuapp.com/api/v1/news';
