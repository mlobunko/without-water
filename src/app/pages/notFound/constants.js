export const errorMessage = 'Page not found';
