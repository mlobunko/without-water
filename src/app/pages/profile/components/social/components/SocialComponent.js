import React from 'react';
import PropTypes from 'prop-types';

import { StyledSocialComponent } from './styles';
import Header from './Header';

export const SocialComponent = ({ social, getIconByName }) => (
  <React.Fragment>
    <Header text={'Social'} />
    <StyledSocialComponent>
      {social.map((el, id) => (
        <div key={id}>
          <a
            href={el.link}
            target="_blank"
            rel="noopener noreferrer"
            aria-label={el.label}
          >
            {getIconByName(el.label)}
          </a>
        </div>
      ))}
    </StyledSocialComponent>
  </React.Fragment>
);

export default SocialComponent;

SocialComponent.propTypes = {
  social: PropTypes.array.isRequired,
  getIconByName: PropTypes.func.isRequired
};
