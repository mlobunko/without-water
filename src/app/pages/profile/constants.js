export const NAME = 'profile';

export const ERROR_TRANSLATE = {
  user_not_found: 'User not found'
};
