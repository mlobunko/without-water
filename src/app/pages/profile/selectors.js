import { createSelector } from 'reselect';

import { NAME } from './constants';
import * as model from './model';
import components from './components';

export const getAll = state => state[NAME];

export const getCity = state => state[NAME][components.city.constants.NAME];

export const getLanguages = state =>
  state[NAME][components.languages.constants.NAME];

export const getSocial = state => state[NAME][components.social.constants.NAME];

export const getSocialWebFirst = createSelector([getSocial], social =>
  model.placeWebIconFirst(social)
);
