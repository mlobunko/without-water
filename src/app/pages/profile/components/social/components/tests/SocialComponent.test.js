import React from 'react';
import { shallow } from 'enzyme';

import { SocialComponent } from '../SocialComponent';
import { NAME } from '../../../../constants';
import * as constants from '../../constants';
import * as model from '../../../../model';
import { getSocialWebFirst } from '../../../../selectors';

describe('render', () => {
  const state = {
    [NAME]: {
      [constants.NAME]: [
        { label: 'vk', link: '123' },
        { label: 'web', link: '321' }
      ]
    }
  };
  it('renders correctly', () => {
    const wrapper = shallow(
      <SocialComponent
        getIconByName={model.getIconByName}
        social={getSocialWebFirst(state)}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
