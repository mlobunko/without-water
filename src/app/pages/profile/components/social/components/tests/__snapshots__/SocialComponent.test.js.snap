// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<Fragment>
  <Header
    text="Social"
  />
  <StyledSocialComponent>
    <div
      key="0"
    >
      <a
        aria-label="web"
        href="321"
        rel="noopener noreferrer"
        target="_blank"
      >
        <StyledSocialComponent
          icon={
            Object {
              "icon": Array [
                496,
                512,
                Array [],
                "f0ac",
                "M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z",
              ],
              "iconName": "globe",
              "prefix": "fas",
            }
          }
        />
      </a>
    </div>
    <div
      key="1"
    >
      <a
        aria-label="vk"
        href="123"
        rel="noopener noreferrer"
        target="_blank"
      >
        <StyledSocialComponent
          icon={
            Object {
              "icon": Array [
                576,
                512,
                Array [],
                "f189",
                "M545 117.7c3.7-12.5 0-21.7-17.8-21.7h-58.9c-15 0-21.9 7.9-25.6 16.7 0 0-30 73.1-72.4 120.5-13.7 13.7-20 18.1-27.5 18.1-3.7 0-9.4-4.4-9.4-16.9V117.7c0-15-4.2-21.7-16.6-21.7h-92.6c-9.4 0-15 7-15 13.5 0 14.2 21.2 17.5 23.4 57.5v86.8c0 19-3.4 22.5-10.9 22.5-20 0-68.6-73.4-97.4-157.4-5.8-16.3-11.5-22.9-26.6-22.9H38.8c-16.8 0-20.2 7.9-20.2 16.7 0 15.6 20 93.1 93.1 195.5C160.4 378.1 229 416 291.4 416c37.5 0 42.1-8.4 42.1-22.9 0-66.8-3.4-73.1 15.4-73.1 8.7 0 23.7 4.4 58.7 38.1 40 40 46.6 57.9 69 57.9h58.9c16.8 0 25.3-8.4 20.4-25-11.2-34.9-86.9-106.7-90.3-111.5-8.7-11.2-6.2-16.2 0-26.2.1-.1 72-101.3 79.4-135.6z",
              ],
              "iconName": "vk",
              "prefix": "fab",
            }
          }
        />
      </a>
    </div>
  </StyledSocialComponent>
</Fragment>
`;
